cdap-watchdog/src/main/java/co/cask/cdap/logging/write/AvroFileWriter.java

```diff
@@ -174,9 +182,25 @@ private AvroFile createAvroFile(LoggingContext loggingContext, long timestamp) t
     return avroFile;
   }
 
-  private Location createLocation(String pathFragment, long timestamp) throws IOException {
+  private Location createLocation(LoggingContext loggingContext, long timestamp) throws IOException {
+    Map<String, LoggingContext.SystemTag> systemTags = loggingContext.getSystemTagsMap();
+    LoggingContext.SystemTag topLevelTag;
+    if (systemTags.containsKey(NamespaceLoggingContext.TAG_NAMESPACE_ID)) {
+      topLevelTag = systemTags.get(NamespaceLoggingContext.TAG_NAMESPACE_ID);
```

Comment:
This logic is very specific to `LoggingContext`, I don't think it is a good idea to have it here. Also, see my comment about `getLogPathFragment()`